### PR TITLE
Introduction of Quadric Layers - Action triggered by Ankith Gunapal 

### PR DESCRIPTION
**Pull request by diro5t**
This is a small tutorial for quadric layers as it pertains to the feature list of PyTorch 2.1. Ankith asked me to provide a small tutorial for this.

### CODE REVIEW

File: intermediate_source/quadric_layers_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    all += 1
+
+print("Number of tested images=", all)
+print("\nModel Test Accuracy =", (corrects / all))
+
+######################################################################
+# 6. Conclusion
+# -------------
+#
+# Quadric layers can easily be used to reduce model size in many applications just by replacing linear layers.
```
Comment (agunapal) [inline intermediate_source/quadric_layers_tutorial.py:228]: Can we show the reduction in model size in this tutorial, (before vs after)

File: intermediate_source/quadric_layers_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    for images, labels in train_loader:
+        images = images.view(images.shape[0], -1)
+        optimizer.zero_grad()
+        output = model(images)
+        loss = loss_fn(output, labels)
+        loss.backward()
+        optimizer.step()
+        train_loss += loss.item()
+    else:
+        print("Epoch {} - Training loss: {}".format(e, train_loss/len(train_loader)))
```
Comment (agunapal) [inline intermediate_source/quadric_layers_tutorial.py:182]: Nit: f-strings would be nice

**Review by agunapal — CHANGES_REQUESTED** [review]
@diro5t Thanks for the PR. Looks good overall. It would be great to show the reduction in model size in the tutorial.
Can you please sign the CLA too.

**Comment by agunapal** [comment]
cc: @msaroufim

**Review by msaroufim — APPROVED** [review]
this was fun to learn about thanks! Appreciated that you showed how it was a dropin replacement for `nn.Linear()`

**Comment by diro5t** [comment]
Thanks, I hope a lot of interested people will try it out in all kinds of applications--best regardsDirk RoeckmannFive Troop Consulting Services ***@***.*** Aug 5, 2023, at 8:55 PM, Mark Saroufim ***@***.***> wrote:﻿
@msaroufim approved this pull request.

this was fun to learn about thanks! Appreciated that you showed how it was a dropin replacement for nn.Linear()

**Review by agunapal — APPROVED** [review]
Looks good to me. Thanks

File: intermediate_source/quadric_layers_tutorial.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+# model size: 25434 parameters
+#
+# comparable linear  model:
+# -------------------------
+# model = nn.Sequential(nn.Linear(784, 128),
+#                      nn.ReLU(),
+#                      nn.Linear(128, 10),
+#                      nn.LogSoftmax(dim=1))
+# model size: 101770 parameters
+#
```
Comment (svekars) [inline intermediate_source/quadric_layers_tutorial.py:250]: Please add a conclusion section. For example:

Conclusion
-------------

In this tutorial, you have learned .....<add more text here>
For more information, see: 

* link1
* link2
* link3

File: intermediate_source/quadric_layers_tutorial.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+# Thanks for reading! Any feedback is highly appreciated. Just create an issue
+# `here <https://github.com/pytorch/pytorch/issues>` if you have any.
```
Comment (svekars) [inline intermediate_source/quadric_layers_tutorial.py:252]:
```suggestion
```

**Review by svekars — COMMENTED** [review]
Thanks so much! Some editorial suggestion here. Please let me know if you have any questions. Also please review the following:
* Spellcheck is [failing](https://github.com/pytorch/tutorials/actions/runs/5766975285/job/15635921009?pr=2520). Words like quadric, Quadric can be added to the [en-wordlist.txt](https://github.com/pytorch/tutorials/blob/main/en-wordlist.txt).
*  Need to add your tutorial as a ``customcarditem `` and to the appropriate ``toctree`` in [index.rst](https://github.com/pytorch/tutorials/blob/main/index.rst).
* We will be targeting to run this against the PyTorch 2.1 RC and merging on the day of the 2.1 release in October.

**Review by mikaylagawarecki — CHANGES_REQUESTED** [review]
Hey, it seems like this was prompted by the prototype feature request for torch.nn.Quadric in 2.1 that was submitted for it. 

Let's wait until we have a conclusion on the associated issue  https://github.com/pytorch/pytorch/issues/105365 before we land this tutorial

**Comment by diro5t** [comment]
> Hey, it seems like this was prompted by the prototype feature request for `torch.nn.Quadric` that was submitted for 2.1.
> 
> Let's wait until we have a conclusion on the associated issue [pytorch/pytorch#105365](https://github.com/pytorch/pytorch/issues/105365) before we land this tutorial


Does this mean, I should not apply the proposed changes for now? Please advise

**Pull request closed without merging**